### DISCUSSION TITLE
Reverse Geocode using TIGER 2020 data

### DISCUSSION THREAD

**Discussion by raynerhoward** [osm-search/Nominatim#2313] [Geocoding around the world]
Is it possible to reverse geocode using only the TIGER 2020 data?  In other words, is there a flag I can use in the URL to tell it to use TIGER 2020 but not OSM data?  There are many areas in the U.S. that have poor addressing via OSM.  I'd like to bypass the OSM results and go straight to the TIGER 2020 results.  Is it very processor intensive to reverse geocode against TIGER roads data?

An example:  https://nominatim.openstreetmap.org/reverse?lat=35.0389&lon=-85.3075

Rather than return "CARTA Shuttle", I'd like it to return an actual address.

**Comment by mtmail** [comment]
The TIGER import adds streets and housenumbers (usually house number interpolation lines), for full results including city, state, country you still need OSM data. Try importing only the `admin` style, then import TIGER
https://nominatim.org/release-docs/latest/admin/Import/#filtering-imported-data

**Reply by raynerhoward** [reply]
Tried your suggestion, it only reverse geocodes to the town now.  When searching for an address, it now shows a blank [] result.  It's like the tiger data isn't there, but I added it as before.

I'm building the nominatim database from a small county-level PBF extract (south-eastern Tennessee).  I copied the import-admin.style file into my nominatim-project directory and added 

NOMINATIM_IMPORT_STYLE=import-admin.style

to the .env file.  

Contents of my .env file:

NOMINATIM_DATABASE_DSN="pgsql:dbname=nominatim"
NOMINATIM_DATABASE_WEBUSER="www-data"
NOMINATIM_USE_US_TIGER_DATA=yes
NOMINATIM_IMPORT_STYLE=import-admin.style

Anything else I could try?

**Reply by lonvia** [reply]
If you want addresses, you want the `address` style.

**Reply by raynerhoward** [reply]
That worked, thanks.

**Comment by lonvia** [comment]
I suspect you don't really care if the result is tiger data. Is it possible that what you actually want is to always get a full address back with your reverse request and nothing else? That is currently possible but there is [an open issue](https://github.com/osm-search/Nominatim/issues/1469) for it.

The best workaround at the moment is probably to use the `address` style. If you want to forcably ignore housenumbers from OSM data, use the `street` style. The latter will ignore postcodes as well, though.

**Reply by raynerhoward** [reply]
Address style seems to be the best solution.  It's reverse geocoding to some OSM addresses but if it can't find one, or get close enough to one, it returns an address from Tiger 2020 roads.  Thanks again!
